source/agora/test/Flash.d

```diff
@@ -1072,3 +1072,143 @@ unittest
     alice.waitForUpdateIndex(chan_id, 6);
     bob.waitForUpdateIndex(chan_id, 6);
 }
+
+/// test various error cases
+unittest
+{
+    TestConf conf = { txs_to_nominate : 1, payout_period : 100 };
+    auto network = makeTestNetwork(conf);
+    network.start();
+    scope (exit) network.shutdown();
+    //scope (failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+    scope (failure) node_1.printLog();
+
+    // split the genesis funds into WK.Keys[0] .. WK.Keys[7]
+    auto txs = genesisSpendable().take(8).enumerate()
+        .map!(en => en.value.refund(WK.Keys[en.index].address).sign())
+        .array();
+
+    foreach (idx, tx; txs)
+    {
+        node_1.putTransaction(tx);
+        network.expectHeightAndPreImg(Height(idx + 1), network.blocks[0].header);
+    }
+
+    auto factory = new FlashNodeFactory(network.getRegistry());
+    scope (exit) factory.shutdown();
+    scope (failure) factory.printLogs();
+
+    const alice_pair = Pair(WK.Keys[0].secret, WK.Keys[0].secret.toPoint);
+    const bob_pair = Pair(WK.Keys[1].secret, WK.Keys[1].secret.toPoint);
+
+    // workaround to get a handle to the node from another registry's thread
+    const string address = format("Validator #%s (%s)", 0,
+        WK.Keys.NODE2.address);
+    auto alice = factory.create(alice_pair, address);
+    auto bob = factory.create(bob_pair, address);
+
+    const Settle_1_Blocks = 0;
+
+    // the utxo the funding tx will spend (only relevant to the funder)
+    const utxo = UTXO.getHash(hashFull(txs[0]), 0);
+
+    // error on mismatching genesis hash
+    ChannelConfig bad_conf = { funder_pk : alice_pair.V };
+    auto open_res = bob.openChannel(bad_conf, PublicNonce.init);
+    assert(open_res.error == ErrorCode.InvalidGenesisHash, open_res.to!string);
+
+    // error on capacity too low
+    auto res = alice.openNewChannel(
+        utxo, Amount(1), Settle_1_Blocks, bob_pair.V);
+    assert(res.error == ErrorCode.RejectedFundingAmount, res.to!string);
+
+    // error on capacity too high
+    res = alice.openNewChannel(
+        utxo, Amount(1_000_000_000), Settle_1_Blocks, bob_pair.V);
+    assert(res.error == ErrorCode.RejectedFundingAmount, res.to!string);
+
+    // error on settle time not within limits
```

Comment:
Shouldnt there be a corresponding "too low" test?

Reply:
I'll add it.